pkg/endpoint/types.go

```diff
@@ -36,12 +36,13 @@ type Provider interface {
 
 // Endpoint is a tuple of IP and Port representing an instance of a service
 type Endpoint struct {
-	net.IP `json:"ip"`
-	Port   `json:"port"`
+	net.IP   `json:"ip"`
+	Port     `json:"port"`
+	ProxyUID string `json:"proxy_uid"`
```

Comment:
This looks a bit hacky to me because the ProxyUID is something specific to the implementation of the service mesh, and not necessarily an attribute of a service's endpoint. If we ever were to run the proxy as a DaemonSet, this assumption could break.

I do think it is a simple approach to glue the code though.

Reply:
I love this idea of a unique identifier for an `Endpoint`.
I'd rename `ProxyUID` to perhaps `ID` or `UID` and I think this is good to go. But I don't see anything wrong with `ProxyUID` either - proxies are going to be around for some time.

Reply:
@draychev You are assuming that different endpoint providers will have an ID for an endpoint, similar to the one OSM add derives from labels added to the pod. This may or may not be true.

Reply:
@shashankram I am not assuming - I am proposing. It is up to us to do that. How that ID is generated is yet to be determined. This could be a UUID, or a hash of some kind.

Reply:
> @shashankram Your point 1 is valid. I'd propose we feature flag this and use this when the replica count is 1. (We need to continue to work on HA and higher replica count but this won't happen quickly and not in this PR.)
> 
> On point 2. A proxy being connected is a good indication that it is also healthy and most likely has latest configuration. A proxy that is not connected - wonder why - but also config will go stale. Could unconnected proxy also be a rogue pod? I'd say an Envoy that connected at least once within the last X minutes is acceptable to be used as an endpoint. An Envoy that never connected should not be sent as an Endpoint.
> 
> Point 3 - let's find a way to resolve this. If the filter introduced in this PR is behind a feature flag (solves #1 temporarily) - I'd be supportive of merging this and digging into #2 and #3

@draychev I am not convinced we should be merging this proposal just yet, given the tradeoffs indicated by @nojnhuh in https://docs.google.com/document/d/1M1-lKRIeZ53sKen1tfHG8Q9Etof6PHSVyFcu1o0oJGQ/edit# suggest it might not be worth adopting at the moment:
```
Overall, given this approach only adds complexity without reducing it elsewhere and could introduce some serious issues, if the only meaningful change is really lower error latency in some (maybe rare) cases, I’m not confident this approach is worth adopting now.
```

At this moment I don't see any noticeable benefits with this proposal, I could be missing something though. @draychev , @nojnhuh would you mind summarizing the benefits of this change? What has it improved? Could you quantify the improvement?